test/test_framework.py

```diff
@@ -514,26 +549,20 @@ def on_state_changed(self, event):
         self.assertEqual(obj_copy.state.foo, 42)
         self.assertEqual(obj_copy.state.bar, "s")
 
-        # But it has observed no changes since instantiation:
-        self.assertEqual(obj_copy.changes, 0)
+        self.assertEqual(list(obj.state.list), [1, 2])
+        self.assertEqual(dict(obj_copy.state.dict), {1: "a", 2: "b"})
+        self.assertEqual(set(obj_copy.state.set), set([1, 2, b'\x00']))
```

Comment:
The tests for the richer types feel somewhat dull. I haven't run it, but I believe they could literally be dumb values sticked into the instance and everything would still pass. I suggest keeping this test, which is a basic_test per its name, just doing the foundational logic of checking that commits work well, etc, and then checking the richer values more deeply into the test we have specific for that at the bottom (mutable_types or something similar).